federation/normalize.go

```diff
@@ -115,7 +115,14 @@ func (f *flattener) flattenFragments(selectionSet *graphql.SelectionSet, typ *gr
 
 	// Descend into fragments matching the current type.
 	for _, fragment := range selectionSet.Fragments {
-		ok, err := f.applies(typ, fragment)
+		ok, err := graphql.ShouldIncludeNode(fragment.Directives)
+		if err != nil {
+			return err
```

Comment:
Can we oop.wrap the error here and in planner.go

Reply:
fixed!